src/redux/Topic/saga.js

```diff
@@ -87,28 +87,37 @@ export function* calculateWinningsHandler() {
     try {
       const {
         contractAddress,
-        senderAddress,
+        walletAddresses,
       } = action.params;
 
-      const options = {
-        method: 'POST',
-        body: JSON.stringify({
-          contractAddress,
-          senderAddress,
-        }),
-        headers: {
-          'Content-Type': 'application/json',
-        },
-      };
+      const value = [];
+      for (let i = 0; i < walletAddresses.length; i++) {
+        const item = walletAddresses[i];
+        const senderAddress = item.address;
+        const options = {
+          method: 'POST',
+          body: JSON.stringify({
+            contractAddress,
+            senderAddress,
+          }),
+          headers: {
+            'Content-Type': 'application/json',
+          },
+        };
+
+        const result = yield call(request, Routes.api.winnings, options);
+        let botWon = 0;
+        let qtumWon = 0;
+
+        if (result) {
+          botWon = satoshiToDecimal(result['0']);
+          qtumWon = satoshiToDecimal(result['1']);
+        }
 
-      const result = yield call(request, Routes.api.winnings, options);
-      let value;
-      if (result) {
-        const botWon = satoshiToDecimal(result['0']);
-        const qtumWon = satoshiToDecimal(result['1']);
-        value = { botWon, qtumWon };
-      } else {
-        value = { botWon: 0, qtumWon: 0 };
+        // return only winning addresses
+        if (botWon || qtumWon) {
```

Comment:
is this true if botWon === 0 or qtumWon === 0?

Reply:
do you mean if botWon === 0 and qtumWon === 0? if botWon === 0 and qtumWon === 0 it will return false.

Reply:
k. just wanted to clarify that it checks botWon > 0 || qtumWon > 0 👍